### PR TITLE
♻️  `webserver`: fixes mypy issues in `director_v2` plugin

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
- ♻️ refactors ``director_v2`` plugin in webserver:

## Related issue/s

- https://github.com/ITISFoundation/osparc-simcore/issues/4071
- https://git.speag.com/oSparc/osparc-ops-environments/-/issues/570

## How to test

```cmd
cd services/web/server
make mypy | grep director_v2
```
## DevOps Checklist

- [x] Check with Andrei commented test
- [x] Check code smells

### CODE REVIEW

File: services/web/server/src/simcore_service_webserver/director_v2/_abc.py
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/director_v2/_abc.py:1]: Are you sure this module `abc` is not public? the idea is to create an abstract interface that can be implemented by external plugins

**Review by pcrespov — APPROVED** [review]

**Review by pcrespov — CHANGES_REQUESTED** [review]
Fix code smells detected by sonarcloud please :-). (check with me if you have any questions)

File: services/web/server/src/simcore_service_webserver/director_v2/_handlers.py [hunk 1]
```diff
@@ -29,8 +29,8 @@
 
 
 class RequestContext(BaseModel):
-    user_id: UserID = Field(..., alias=RQT_USERID_KEY)
-    product_name: str = Field(..., alias=RQ_PRODUCT_KEY)
+    user_id: UserID = Field(..., alias=RQT_USERID_KEY)  # type: ignore
+    product_name: str = Field(..., alias=RQ_PRODUCT_KEY)  # type: ignore
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/director_v2/_handlers.py:33]: why the type ignore here?

File: services/web/server/src/simcore_service_webserver/director_v2/_core_base.py [hunk 2]
```diff
@@ -53,17 +53,21 @@ def _get_exception_from(
     return DirectorServiceError(status=status_code, reason=reason, url=url)
 
 
+class RetryFailedError(Exception):
+    pass
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/director_v2/_core_base.py:57]: Is this used somewhere?

File: services/web/server/tests/unit/with_dbs/03/garbage_collector/test_resource_manager.py [hunk 3]
```diff
@@ -636,17 +642,18 @@ async def mocked_notification_system(mocker):
...
+        #     UserRole.GUEST,
+        #     False,
+        #     marks=pytest.mark.xfail(
+        #         reason=(
+        #             "GC is wrongly assuming that only 1 tab can be opened for GUEST users. "
+        #             "SEE: https://github.com/ITISFoundation/osparc-simcore/issues/3770"
+        #         )
+        #         #
+        #     ),
+        # ),
```
Comment (GitHK) [inline services/web/server/tests/unit/with_dbs/03/garbage_collector/test_resource_manager.py:655]: Why was this changed? Form my knowledge the issue is still there.

File: services/web/server/src/simcore_service_webserver/director_v2/_core_base.py (hunk 2 above)
Comment (GitHK) [reply services/web/server/src/simcore_service_webserver/director_v2/_core_base.py:57]: I see it is only defined here and not used.

**Review by GitHK — COMMENTED** [review]
Please check my comments

File: services/web/server/src/simcore_service_webserver/director_v2/_handlers.py (hunk 1 above)
Comment (matusdrobuliak66) [reply services/web/server/src/simcore_service_webserver/director_v2/_handlers.py:33]: error: Required dynamic aliases disallowed  [pydantic-alias] 
(I think type: ignore is acceptable here)

File: services/web/server/src/simcore_service_webserver/director_v2/_core_base.py (hunk 2 above)
Comment (matusdrobuliak66) [reply services/web/server/src/simcore_service_webserver/director_v2/_core_base.py:57]: good catch thats removing

File: services/web/server/src/simcore_service_webserver/director_v2/_abc.py
Comment (matusdrobuliak66) [reply services/web/server/src/simcore_service_webserver/director_v2/_abc.py:1]: It seems, that **api.py** is the module to access all plugins functionality including abc:
``` api.py
from ._abc import (
    AbstractProjectRunPolicy,
    get_project_run_policy,
    set_project_run_policy,
)
```

File: services/web/server/src/simcore_service_webserver/director_v2/_handlers.py (hunk 1 above)
Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/director_v2/_handlers.py:33]: Have you tried with Literal ?

**Review by sanderegg — COMMENTED** [review]
Thanks! please check the comments.

**Review by sanderegg — APPROVED** [review]
Thanks! please check the comments.

File: services/web/server/tests/unit/with_dbs/03/garbage_collector/test_resource_manager.py (hunk 3 above)
Comment (matusdrobuliak66) [reply services/web/server/tests/unit/with_dbs/03/garbage_collector/test_resource_manager.py:649]: Hey Andrei, I didn't do any "logical" changes, just rearranged/renamed the files. I think it might be connected with the timing that is included in the test `await asyncio.sleep(SERVICE_DELETION_DELAY + 1)` because when i am changing this parameter based on that it ether passes or fails. We can have a look tomorrow together.

Comment (GitHK) [reply services/web/server/tests/unit/with_dbs/03/garbage_collector/test_resource_manager.py:649]: So this is the correct state of the test. feel free to remove the comment above

**Review by GitHK — APPROVED** [review]

**Comment by matusdrobuliak66** [comment]
> Fix code smells detected by sonarcloud please :-). (check with me if you have any questions)

Yes, I would be glad to have a short discussion about some of the code smells.

**Review by pcrespov — APPROVED** [review]
